ContractsApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs

```diff
@@ -252,6 +252,30 @@ public async Task PatchContractThrowsConflictVersionError()
                 Times.Never());
         }
 
+        [Fact]
+        public async Task PatchContractThrowsErrorIfTryingToSuspendBlock()
+        {
+            var today = DateTime.Today;
+            var tomorrow = today.AddDays(1);
+            var currentContract = _fixture.Build<ContractDb>().With(x => x.VersionNumber, (int?) null).With(x => x.StartDate, (DateTime?) today).Create();
+
+            await InsertDataIntoDynamoDB(currentContract).ConfigureAwait(false);
+
+            var contractId = currentContract.Id;
+            var request = _fixture.Create<EditContractRequest>();
+            var suppliedVersion = 0;
+            request.HandbackDate = tomorrow;
+            request.SuspensionDate = tomorrow;
+            request.ContractManagement.AssetHierarchy = AssetHierarchy.Block;
```

Comment:
I guess we have to be careful about scope creep here, but I think the responsibility of locking down the block contract suspension lies with the API and not the client. Client shouldn't have to tell what the contract type is. This approach is understandable with the current state of the data and the API, but it might be worth adding complete support for this new structure in this ticket. 

For example when new contract is created the contract management details should be populated at that time. Then when patching the API would use the data to work out what's allowed and what's not. If we want to decouple the asset hierarchy from the contract hierarchy then the client should be responsible for telling the API what type of conract should be created.

Reply:
As per comment below, we have agreed that the ContractManagement should never be patched. I've adapted the Gateway and its tests to reflect this.